src/Makefile.am

```diff
@@ -8,7 +8,9 @@
 SUFFIXES = .cpp .h .yy .ll .cc .hh .hpp
 
 bin_PROGRAMS = souffle souffle-mcpp souffle-profile
- 
+
+BUILT_SOURCES = parser.hh stack.hh
```

Comment:
main.cpp also depends on ParserDriver.h, and ParserDriver.h depends on parser.hh. The transitive dependency isn't picked up by make, so it can try to build main.cpp using ParserDriver.h before parser.hh is generated. Adding parser.hh as a dependency of main.cpp would fix that, but then the Makefile has to be kept updated if the relevant includes change, which also seemed hacky.

```main.cpp ParserDriver.cpp : parser.hh``` might still be preferable to BUILT_SOURCES, though - the lesser hack.